extensions/ql-vscode/test/unit-tests/common/files.test.ts

```diff
@@ -417,3 +459,125 @@ describe("walkDirectory", () => {
     expect(files.sort()).toEqual([file1, file2, file3, file4, file5, file6]);
   });
 });
+
+describe("findCommonParentDir", () => {
+  const rootDir = parse(process.cwd()).root;
+
+  it("should fail if not all paths are not absolute", async () => {
+    const paths = [
+      join("foo", "bar", "baz"),
+      join("/foo", "bar", "qux"),
+      join("/foo", "bar", "quux"),
+    ];
+
+    expect(() => findCommonParentDir(...paths)).toThrow(
+      "All paths must be absolute",
+    );
+  });
+
+  it("should fail if no path are provided", async () => {
+    expect(() => findCommonParentDir()).toThrow(
+      "At least one path must be provided",
+    );
+  });
+
+  it("should find the common parent dir for multiple paths with common parent", () => {
+    const paths = [
+      join("/foo", "bar", "baz"),
+      join("/foo", "bar", "qux"),
+      join("/foo", "bar", "quux"),
+    ];
+
+    const commonDir = findCommonParentDir(...paths);
+
+    expect(commonDir).toEqualPath(join("/foo", "bar"));
+  });
+
+  it("should return empty path if paths have no common parent", () => {
+    const paths = [
+      join("/foo", "bar", "baz"),
```

Comment:
I think the reason for the test failures on windows is because `join('/foo', 'bar', 'baz')` returns `\foo\bar\baz` without a drive letter, but then we're expecting the common parent dir to be `D:\`.

Would it work to use `join(root, 'foo', 'bar', 'baz')` instead? This will also apply to every other place we use `/` as a prefix to the path.

Reply:
Taking https://github.com/github/vscode-codeql/pull/3289#discussion_r1471396830 has made the tests go green 🎉